modules/extensions-support/src/test/java/org/mule/module/extension/internal/ExtensionOfTestCase.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.hamcrest.Matchers.emptyIterable;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+import static org.mule.module.extension.HeisenbergExtension.EXTENSION_DESCRIPTION;
+import static org.mule.module.extension.HeisenbergExtension.EXTENSION_NAME;
+import static org.mule.module.extension.HeisenbergExtension.EXTENSION_VERSION;
+import org.mule.extension.annotations.Extensible;
+import org.mule.extension.annotations.ExtensionOf;
+import org.mule.extension.annotations.Operation;
+import org.mule.extension.annotations.Operations;
+import org.mule.extension.introspection.declaration.fluent.Declaration;
+import org.mule.extension.introspection.declaration.fluent.OperationDeclaration;
+import org.mule.module.extension.internal.capability.metadata.ExtendingOperationCapability;
+
+import java.util.Set;
+
+import org.junit.Test;
+
+public class ExtensionOfTestCase extends AbstractAnnotationsBasedDescriberTestCase
```

Comment:
This name seems very weird. I would change it for something like ExtensibleExtensionTestCase but better